### DISCUSSION TITLE
styled component SVG resizing bug

### DISCUSSION THREAD

**Discussion by ryliecc** [neuefische/questions#628] [Web] [answered]
_english version below_

Hallo,
ich arbeite gerade am Styling meiner bisher erstellten Pages und Komponenten.
Den aktuellen Stand habe ich commited und gepusht, er ist [hier](https://github.com/ryliecc/capstone_webdev_wanikani-app/blob/styling/components/SessionButton.js) zu finden. Der branch heißt `styling`.

Wenn ich versuche die Größe des SVG Elements auf der SessionButton.js Komponente zu ändern, verändert dies die Größe und das positioning der ButtonSpans, bzw. des gesamten Umfelds, aber nie die Größe des SVGs. Ich habe sämtliche Attribute ausprobiert (viewBox, font-size, width & height, weight...), ich habe es mit einem Wrapper Element versucht, ich habe sämtliche Lösungen aus dem Netz probiert ("padding-bottom hack"...), die turtle gang gefragt und ChatGPT, nirgendwo fand sich eine Lösung für mein spezifisches Problem.

Momentan sind die SVG Grafiken nur Platzhalter, sie sollen aber irgendwann ausgetauscht werden mit anderen Illustrationen. Es hält mich also jetzt gerade nicht auf, muss aber definitiv früher oder später gelöst werden.

Danke schonmal fürs drüber schauen und Kopf zerbrechen 🤗

--------

Hey,
I am currently working on the styling of my previously created pages and components.
I commited and pushed my last version, you can find it [here.](https://github.com/ryliecc/capstone_webdev_wanikani-app/blob/styling/components/SessionButton.js) The branch is called `styling`.

If I try to change the size of my svg element inside the SessionButton.js component, it changes the sizing and positioning of my ButtonSpans or even the whole environment (page), but never the size of the actual svg. I tried any and all possible attributes (viewBox, font-size, width & height, weight...), I tried a wrapper element, I tried any solution I found online (e.g. the "padding-bottom hack" and more), I asked my group and I asked ChatGPT. No one could help me solve my specific problem.

At the moment these svg graphics are just placeholders for other illustrations to come, so that problem does not necessarily block my workflow right now, but it is pretty annoying considering I need to fix it eventually anyway.

Thank you in advance for taking a look at my code 🤗

**Comment by mpagels** [comment, marked as answer]
Would that be an option to use the way the React template is using the `svg`? 
See following code

```js
import React from 'react';
{/*images*/}
import ReactLogo from './logo.svg';

const App = () => {
  return (
    <div className="App">
      <img src={ReactLogo} alt="React Logo" />
    </div>
  );
}
export default App;
```

The styling would be a lot easier, I think.
